contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -2836,3 +2837,16 @@ tsql_printTypmod(const char *typname, int32 typmod, Oid typmodout)
 	return res;
 }
 
+char*
+print_collation_name(Oid collOid)
+{
+	char* res = NULL;
+	char* translated_res = NULL;
+	res = generate_collation_name(collOid);
+	translated_res = tsql_translate_collation_bbf(res);
```

Comment:
if res is null then only try to reverse translate it.

Reply:
When res is not NULL, only then we reverse translate it.